### DISCUSSION TITLE
“Failed to commit changes: The change you're trying to make would violate the required relation"

### DISCUSSION THREAD

**Discussion by lumenwrites** [prisma/prisma#7749] [Q&A] [answered]
I want a post to have several files attached to it. My Prisma models look like this:
```prisma
    model Post {
      id String @id @default(cuid())
      title String
      body String
      files File[] @relation("FilePost")
    }
    
    model File {
      id String @id @default(cuid())
      post Post @relation("FilePost", fields: [postId], references: [id])
      postId String
      name String
      url String
    }
```

Creating posts with files works fine, but when I'm trying to delete the posts that have files attached to them in Prisma Studio, I'm getting the following error:

<img width="470" alt="Screenshot 2021-06-18 at 14 42 53" src="https://user-images.githubusercontent.com/4761235/122570639-d59aa480-d043-11eb-9212-7a40d68ff8d7.png">

Can you help me to figure out what's going on? Am I doing something wrong?

**Comment by ryands17** [comment, marked as answer]
This is happening because of cascade deletes that we will be fixing soon: https://github.com/prisma/prisma/issues/2057

In the meantime, you could perform either of the following workarounds:

1. Delete the related records first and then delete the main record
2.  Make your relation optional:

```prisma
model Post {
      id String @id @default(cuid())
      title String
      body String
      files File[] @relation("FilePost")
    }
    
    model File {
      id String @id @default(cuid())
      // ? means the relation is optional
      post Post? @relation("FilePost", fields: [postId], references: [id])
      postId? String
      name String
      url String
    }
```

And then you can delete the main record without errors.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
